### ISSUE TITLE
Silently inaccurate GPOS subtable with sparse master.

### ISSUE THREAD

**Issue by skef** [fonttools/fonttools#2851]
Consider: [palt-sparse.zip](https://github.com/fonttools/fonttools/files/9804282/palt-sparse.zip)

Regular and Wide span the single Width axis. "Sparse" is a master in-between with fewer glyphs. 

`palt` in the `feature.fea` files has positions for two glyphs in the complete masters but one in Sparse because the other glyph isn't defined there.

The intention in this kind of case will almost always be that the value for the glyph not in the sparse master be treated as absent, so that the two values at the edges interpolate. What varLib does is add a zero at that point because of the aliasing between a genuinely missing value and a zero removed by a static feature compiler.

The relevant code for the single position case starts at line 310 of varLib/merger.py:

```
    for i,values in enumerate(padded):
        for j,glyph in enumerate(glyphs):
            if values[j] is not None: continue
            # Fill in value from other subtables
            # Note!!! This *might* result in behavior change if ValueFormat2-zeroedness
            # is different between used subtable and current subtable!
            # TODO(behdad) Check and warn if that happens?
            v = _Lookup_SinglePos_get_effective_value(merger, merger.lookup_subtables[i], glyph)
            if v is None:
                v = otBase.ValueRecord(valueFormat)
            values[j] = v
```
There is similar code for the pair positioning case.

I'm filing this as an issue rather than a PR because I'm not fully confident about what should happen here. My sense is that the varLib merging code isn't designed to address sparseness in general  so a *fix* would be out of scope.

If that's right, then perhaps the block inside `if v is None:` should check to see if `glyph` is part of that master (`self.ttfs[i]`?) and if not either error out or issue a very stern warning.

What do folks think?

**Comment by behdad** [comment]
You are correct that sparse GPOS merging is not in the design of the merger. That would be a much harder problem.

**Comment by skef** [comment]
If varLib had absolutely *no* provisions for sparseness then this might fall into one big "probably won't work" bucket, even with no warning.

But that isn't really the current state of things. The code at line 1072, for example, supports cases in which only some masters have data for a particular table. So over time various accommodations for easy-to-handle cases have crept in, and some users will inevitably try things with the code looking for what produces a warning or error and what doesn't. Something like that is what turned up this case.

Therefore I think (at least) a warning would be sensible.

**Comment by behdad** [comment]
> If varLib had absolutely _no_ provisions for sparseness then this might fall into one big "probably won't work" bucket, even with no warning.
> 
> But that isn't really the current state of things. The code at line 1072, for example, supports cases in which only some masters have data for a particular table.

The merger code assumes that all masters have the same glyphset. That's why a missing glyph/pair is assumed to have a zero value. That level of sparseness is required to be supported. Different glyphsets is a different beast altogether.
